### PR TITLE
[Kernel][Defaults] Support reading timestamp_ntz stored as INT96

### LINKED ISSUE #2908: [Kernel] Support reading `timestamp_ntz` stored as INT96
Context is [here](https://github.com/delta-io/delta/pull/2907#issuecomment-2062121743).

Basically use [`TimestampConverters.createTimestampConverter`](https://github.com/delta-io/delta/blob/master/kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/TimestampConverters.java#L49) for both timestamp and timestamp_ntz columns.

For testing: use Spark to generated a golden table using the utilities in `GoldenTables.scala`.

### PR DESCRIPTION
**Pull request by raveeram-db**
#### Which Delta project/connector is this regarding?

- [ ] Spark
- [ ] Standalone
- [ ] Flink
- [x] Kernel
- [ ] Other (fill in here)

## Description

Resolves #2908 
## How was this patch tested?

Added a unit test that reads an INT96 column written by Spark as TIMESTAMP_NTZ

## Does this PR introduce _any_ user-facing changes?
No

### CODE REVIEW

File: kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/TimestampConverters.java [hunk 1]
```diff
@@ -90,6 +90,12 @@ public static Converter createTimestampConverter(int initialBatchSize, Type type
     public static Converter createTimestampNtzConverter(int initialBatchSize, Type typeFromFile) {
```
Comment (vkorukanti) [inline kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/TimestampConverters.java:90]: can this be merged into the `createTimestampConverter` method above? Looks like the code is exactly same except in couple of lines.

**Review by vkorukanti — COMMENTED** [review]
LGTM, thank you for the PR!

File: kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/TimestampConverters.java (hunk 1 above)
Comment (raveeram-db) [reply kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/TimestampConverters.java:90]: Yup, that makes sense. I'll update.

Comment (raveeram-db) [reply kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/TimestampConverters.java]: thanks Venki, merged them to a common method

**Review by vkorukanti — APPROVED** [review]

**Pull request merged**